### ISSUE TITLE
What chip does my controller board have?

### ISSUE THREAD

**Issue by spajk1990** [themaddoctor/linux-mybook-tools#21]
Hi,

first of all thank you for helping everybody here.

I have a WD My Book 4T drive (product number is WDBFJK0040HBK).
It is password protected and I know the password.
One day my computer just didn't react when I connected the My Book. Usually I would connect it and enter the password.... But now, nothing. The drive doesn't spin, nothing happens on the computer (no detected hardware changes...)

Since the warranty has expired I opened it, removed the controller board, and inserted it into my docking stations for HDD (it is a Thermaltake BlacX 5G, product number ST0019). I was thinking just to manually start the WD security software and access my data...
The drive immediately started spinning. New hardware was detected... I got a message that the drive has to be initialize to be used. This is when I remembered that the drive uses hardware encryption and that I need the controller board for that. Of course I canceled the initialize message.

Then I got the idea to order a new controller board. My controller board is 4060-705149-000 REV A.
I found a used and working one on ebay and ordered it. It will take 2 weeks to arrive.

Then I found your thread here on github... But I can't figure out what chip is my board using. I have this chips with writing on it:

-winbond 25X20CLVIG 1434
-asmedia ASM1051W BNF25027B3 1436
-RT828 GSP1J006
-APM4532 RQ9BS

I assume the Asmedia one is the one I am looking for.
Can I use this method If I have this chip? Also can I use my Raspberry Pi3 and the docking station I mentioned above for this?

I haven't tried anything else since I connected it to windows, since I read on your post that even though I haven't initialized the disk on windows it still probably has messed up the MBR... 

So what do you think I should do next? Wait for the controller board to arrive or try your method?

Regards,

Spajk

**Comment by themaddoctor** [comment]
The ASM1051W does not do encryption, as far as I know.

If the keys are stored on the ROM chips on the board, you will have to do some swapping.

Can you connect the disk to a linux computer and dump the first two MB for me to look at?
The command is "dd if=/dev/sdX count=4096 of=dump.bin" where you have to replace "sdX"
with the correct thing for your disk. See my PDF in this project on how to determine that.

My method won't work for that chip, by the way.

**Comment by spajk1990** [comment]
Thank you for the quick reply. I will do it on Saturday and send you the dump.

**Comment by Animus120** [comment]
Hello, I'm currently having troubles right now. I have the same product number for WD MyBook WDBFJK0040HBK but I probably have different controller (it writes 4061-705149-A00 Rev. AD) yet I cannot figure out which chip I have.
The writing on the controller are:
ELEC-8 E3330BM
33:5 94V-0
5H6C
150603C

**Comment by Animus120** [comment]
Okay, I have the ASMEDIA ASM1051W chip.
What do I do now? I have plugged the HDD to a linux PC but I cannot figure out how to decrypt it so I can copy the inside of it because I want to recover the data. I never set any password at all on it.

**Comment by themaddoctor** [comment]
I don't believe that that chip does encryption. You probably have a mismatch with the block sizes. Do "fdisk -l /dev/sd_" where _ is the right thing for your disk. If the block size is anything other than 512, then you may have to repartition it.

**Comment by themaddoctor** [comment]
(it will say "sector size")

**Comment by Animus120** [comment]
Okay, I just did what you said and I got this:
https://my.mixtape.moe/ktwwcx.png
So that means I have to repartition it?

**Comment by themaddoctor** [comment]
Sorry, your server is misconfigured and I can't get a secure connection. Why don't you just copy the output into a comment?

**Comment by themaddoctor** [comment]
Try mounting it and reading your data.

**Comment by Animus120** [comment]
I did, but it didn't detect as NTFS partition, so Linux won't mount it.
I'm on Ubuntu 18.04 if that matters.

**Comment by themaddoctor** [comment]
what does "sudo file -s /dev/sdb1" tell you?

**Comment by Animus120** [comment]
it says "data"
That's it.

**Comment by Animus120** [comment]
Should I take screenshot of it, just in case?
I haven't plugged it into Windows, I put my HDD straight into Linux but it only reads like that.

**Comment by themaddoctor** [comment]
dump the first 2MB and upload them as a zipfile

sudo dd if=/dev/sdb count=4096 of=dump.bin

**Comment by Animus120** [comment]
[dump.zip](https://github.com/themaddoctor/linux-mybook-tools/files/2665667/dump.zip)
Here you go, sorry for the late reply. Seems like we have different timezone.

**Comment by themaddoctor** [comment]
/dev/sdb1 256 976746239 976745984 465.8G 7 HPFS/NTFS/exFAT
                  ^^^
See this 256? It is supposed to be 2048. And 976746239 should be 7813969913 (or larger, to be safe). I guess 'fdisk' was reporting 512-byte sectors but adjusting the number elsewhere.

You need to fix your partition table, like I was saying. Be careful, and use fdisk's built-in help.

**Comment by Animus120** [comment]
Thank for the pointer, I can access my disk using testdisk.
Thanks a lot man.

**Comment by Zibri** [comment]
The weird thing with ELEC-8 E3330BM is this:
it works with my drive, but if I connect any other drive (that works with a normal usb/sata interface) windows asks me to partition it. But the drive size/make/model is correctly detected.
On the board, all relevant chips are black, no writings whatsoever.

**Comment by paugarube** [comment]
> Thank for the pointer, I can access my disk using testdisk. Thanks a lot man.

@Animus120 please! Can you explain in detail how did you do with disk drill to solve the access?
I have the same disk and the same access problem.
Thanks in advance, for @themaddoctor too!
